### PR TITLE
build script to build goconserver

### PR DESCRIPTION
**Pull request by xuweibj**
build script to build goconserver

Call ``make deps``, ``make rpm|makedeb`` based on OS and ARCH.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+function check_linux_distro()
+{
+    local distro="$(source /etc/os-release >/dev/null 2>&1 && echo "${ID}")"
+    [[ -z "${distro}" && -f /etc/redhat-release ]] && distro="rhel"
+    [[ -z "${distro}" && -f /etc/SuSE-release ]] && distro="sles"
+    echo "${distro}"
+}
+
+build_dir=/goconserver_build
```
Comment (zet809) [inline build.sh]: An easier way for line 11-14 is `build_dir=${DEST:-/goconserver_build}`

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+if [ ${DEST} ]; then
+    build_dir=${DEST}
+fi
+mkdir -p $build_dir
+
+XCAT_BUILD_DISTRO="$(check_linux_distro)"
+echo "[INFO] Start to build goconserver on $XCAT_BUILD_DISTRO"
+
+cur_path=$(cd "$(dirname "$0")"; pwd)
```
Comment (zet809) [inline build.sh]: Why not cd `dirname "$0"` directly?

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+build_dir=/goconserver_build
+if [ ${DEST} ]; then
+    build_dir=${DEST}
+fi
+mkdir -p $build_dir
+
+XCAT_BUILD_DISTRO="$(check_linux_distro)"
+echo "[INFO] Start to build goconserver on $XCAT_BUILD_DISTRO"
```
Comment (robin2008) [inline build.sh]: There are too much `goconserver` in this script, could you use a variable for it,  so that all the below scripts are common

**Comment by zet809** [comment]
Good to me, I agree to merge.

**Pull request merged**